domain/src/main/java/org/oppia/domain/oppialogger/LogReportStorageHelper.kt

```diff
@@ -0,0 +1,145 @@
+package org.oppia.domain.oppialogger
+
+import androidx.lifecycle.LiveData
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import org.oppia.app.model.EventLog
+import org.oppia.app.model.ExceptionLog
+import org.oppia.app.model.OppiaEventLogs
+import org.oppia.app.model.OppiaExceptionLogs
+import org.oppia.data.persistence.PersistentCacheStore
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.data.DataProviders
+import org.oppia.util.logging.Logger
+import org.oppia.util.threading.BackgroundDispatcher
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class LogReportStorageHelper @Inject constructor(
+  private val cacheStoreFactory: PersistentCacheStore.Factory,
+  private val dataProviders: DataProviders,
+  private val logger: Logger,
+  @BackgroundDispatcher private val backgroundCoroutineDispatcher: CoroutineDispatcher,
+  @EventLogStorageCacheSize private val eventLogStorageCacheSize: Int,
+  @ExceptionLogStorageCacheSize private val exceptionLogStorageCacheSize: Int
+) {
+  private enum class LogReportingCase {
+    EVENT_LOG,
+    EXCEPTION_LOG
+  }
+  private val eventLogStore =
+    cacheStoreFactory.create("event_logs", OppiaEventLogs.getDefaultInstance())
+  private val exceptionLogStore =
+    cacheStoreFactory.create("exception_logs", OppiaExceptionLogs.getDefaultInstance())
+  private val coroutineScope = CoroutineScope(backgroundCoroutineDispatcher)
+
+  fun addEventLog(eventLog: EventLog) {
+    coroutineScope.launch {
+      checkStoreCacheStatus(
+        eventLogStore.readDataAsync().await().serializedSize.toByte(),
```

Comment:
You are checking the status with the current store, but not with the store + the size of the new event that we want to add. So it will only delete an entry from the store after it has already gone above the cache limit

Reply:
Yes, modified. PTAL.